src/kaboom.ts

```diff
@@ -422,15 +422,22 @@ const COMP_EVENTS = new Set([
 	"inspect",
 ]);
 
+type UnionToIntersection<U> =
+  (U extends any ? (k: U) => void : never) extends ((k: infer I) => void) ? I : never
+type Defined<T> = T extends any ? Pick<T, { [K in keyof T]-?: T[K] extends undefined ? never : K }[keyof T]> : never;
+type Expand<T> = T extends infer U ? { [K in keyof U]: U[K] } : never;
```

Comment:
Credit to the answer here: https://stackoverflow.com/questions/62366377/how-to-type-generic-merging-function-in-typescript for the TypeScript magic.